plugins/modules/ipbase_info.py

```diff
@@ -0,0 +1,163 @@
+        "country_code": "AT",
+        "hostname": "81-223-100.100.static.upcbusiness.at",
+        "ip": "81.223.100.100",
+        "is_in_european_union": true,
+        "latitude": 48.1861686706543,
+        "longitude": 16.403240203857422,
+        "region": "Vienna",
+        "region_code": "AT-9",
+        "timezone": "Europe/Vienna",
+        "zip": "1030"
+    }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible.module_utils.urls import fetch_url
+
+
+USER_AGENT = 'ansible-ipbase-module/0.1.0'
+BASE_URL = 'https://api.ipbase.com/v2/info?hostname=1'
+
+
+class IpbaseInfo(object):
+
+    def __init__(self, module):
+        self.module = module
+
+    def _get_url_data(self, url):
+        response, info = fetch_url(
+            self.module,
+            url,
+            force=True,
+            timeout=10,
+            headers={
+                'Accept': 'application/json',
+                'User-Agent': USER_AGENT,
+            })
+
+        if (info['status'] != 200):
+            self.module.fail_json(msg='The API request to ipbase.com returned an error status code {0}'.format(info['status']))
+        else:
+            try:
+                content = response.read()
+                result = self.module.from_json(content.decode('utf8'))
+            except ValueError:
+                self.module.fail_json(
+                    msg='Failed to parse the ipbase.com response: '
+                    '{0} {1}'.format(url, content))
+            else:
+                return result
+
+    def info(self):
+
+        ip = self.module.params['ip']
+        apikey = self.module.params['apikey']
+
+        url = BASE_URL
+        if ip:
+            url += '&ip=' + str(ip)
+
+        if apikey:
+            url += '&apikey=' + str(apikey)
+
+        response = self._get_url_data(url)
+
+        result = dict(
+            ip=response['data']['ip'],
+            hostname=response['data']['hostname'],
+            continent=response['data']['location']['continent']['name'],
+            continent_code=response['data']['location']['continent']['code'],
+            country=response['data']['location']['country']['name'],
+            country_code=response['data']['location']['country']['alpha2'],
+            is_in_european_union=response['data']['location']['country']['is_in_european_union'],
+            region=response['data']['location']['region']['name'],
+            region_code=response['data']['location']['region']['alpha2'],
+            city=response['data']['location']['city']['name'],
+            zip=response['data']['location']['zip'],
+            latitude=response['data']['location']['latitude'],
+            longitude=response['data']['location']['longitude'],
+            timezone=response['data']['timezone']['id'],
+            as_name=response['data']['connection']['organization'],
+            as_number=response['data']['connection']['asn'],
+        )
+
+        return dict(data=result)
```

Comment:
A potentially more sustainable way is to return simply `response['data']`, and document the structure in `RETURN`. Then once `ipbase.com` changes the structure, or adds more output, users will also get hold of that.

Reply:
that makes much more sense!